pkg/osa/types.go

```diff
@@ -0,0 +1,111 @@
+package osa
+
+import (
+	"github.com/ghodss/yaml"
+	"github.com/jim-minter/azure-helm/pkg/api"
+	"github.com/jim-minter/azure-helm/pkg/arm"
+	"github.com/jim-minter/azure-helm/pkg/config"
+	"github.com/jim-minter/azure-helm/pkg/helm"
+	"io/ioutil"
+)
+
+type OSA interface {
+	Validate() error
+	GenerateConfig() (*config.Config, error)
```

Comment:
Should return `[]byte, error`?

Reply:
The updating of config is going to be a pretty important piece as well, so `GenerateConfig` isn't a great name if it's going to do both.  I'm pretty tempted to have two completely different functions in the interface, one for Generate and one for Update.